WordPress/Classes/ViewRelated/System/Notices/NoticeView.swift

```diff
@@ -202,21 +215,46 @@ class NoticeView: UIView {
         cancelBackgroundView.addTopBorder()
         cancelBackgroundView.addTrailingBorder()
 
-        actionButton.titleLabel?.font = notice.style.actionButtonFont
+        actionButton.titleLabel?.adjustsFontForContentSizeCategory = true
         actionButton.setTitleColor(.white, for: .normal)
         actionButton.on(.touchUpInside) { [weak self] _ in
             self?.actionButtonTapped()
         }
         actionButton.setContentCompressionResistancePriority(.required, for: .vertical)
 
-        cancelButton.titleLabel?.font = notice.style.cancelButtonFont
+        cancelButton.titleLabel?.adjustsFontForContentSizeCategory = true
         cancelButton.setTitleColor(notice.style.messageColor, for: .normal)
         cancelButton.on(.touchUpInside) { [weak self] _ in
             self?.cancelButtonTapped()
         }
         cancelButton.setContentCompressionResistancePriority(.required, for: .vertical)
     }
 
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if previousTraitCollection?.preferredContentSizeCategory != traitCollection.preferredContentSizeCategory {
+            preferredContentSizeDidChange()
+        }
+    }
+
+    private func preferredContentSizeDidChange() {
+        cancelButton.titleLabel?.font = notice.style.cancelButtonFont
+        actionButton.titleLabel?.font = notice.style.actionButtonFont
+        titleLabel.font = notice.style.titleLabelFont
```

Comment:
To set the font when the dynamic size changes works well when font instance also changes.
In this case It seems that the `style` field and the fonts that it holds are all immutable, so this should not affect (given that the fonts are set somewhere else).

The fonts themselves are not responding properly to content size changes though, so it could be beneficial to make `style` return a new font instance each time.

**Note:** It might be worth it to check `WPStyleGuide` fonts system, and see why they don't change dynamically. But that sounds as a small project of its own (for the future).

Reply:
Great catch @etoledom I have updated the style implementations to return new font instances everytime we access, that seems like solving all our problems.

>Note: It might be worth it to check WPStyleGuide fonts system, and see why they don't change dynamically. But that sounds as a small project of its own (for the future).

 I didn't quite understand this, actually WPStyleGuide creates a new font everytime.

Reply:
It should not be necessary to reset the font every time content size changes in cases where we just need one of the default size categories without custom font weight. They should adapt automatically given that the label's `adjustsFontForContentSizeCategory` is set to `true`. 

If I recall correctly, this was working properly at some point. Maybe we broke it when we added the maximum font size parameter 🤔

Reply:
Oh I see so you are saying, it should be working without creating a new instance.